carbon-design-system/carbon-addons-iot-react
# fix(side-nav): sidenav menu not collapsing

### Merging into `master` to make it into 8.7

Closes #2880 

**Summary**

- the SideNav wasn't collapsing the expanded sections in rail mode, because the display names were missing and that's how carbon is checking what should be collapsed. Manually added displayNames to all carbon components.

**Change List (commits, features, bugs, etc)**

- added displayName to carbon components

**Acceptance Test (how to verify the PR)**

- go to a [side nav story](https://deploy-preview-3214--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-sidenav--side-nav-component)
- hover over the side nav
- open a nested menu
- hover off the side nav
- confirm that the nested menu collapsed in the rail and isn't taking up extra space

**Regression Test (how to make sure this PR doesn't break old functionality)**

- confirm no adverse side-effects with the SideNav.



**Things to look for during review**
